listenbrainz/webserver/static/js/src/BrainzPlayer.tsx

```diff
@@ -186,12 +190,46 @@ export default class BrainzPlayer extends React.Component<
   };
 
   playListen = (listen: Listen, datasourceIndex: number = 0): void => {
+    let selectedDatasourceIndex = datasourceIndex;
     const { onCurrentListenChange } = this.props;
     onCurrentListenChange(listen);
     if (this.firstRun) {
       this.firstRun = false;
     }
-    this.setState({ currentDataSourceIndex: datasourceIndex }, () => {
+
+    /** If available, retreive the service the listen was listened with */
+    const listeningFrom = _get(
```

Comment:
Could we put this in a new function? `sortSourcesByListenData` maybe? I think the `playListen` function is getting a bit too big.

Reply:
Addressed in PR #904